### ISSUE TITLE
Android Crashing on Fence Add when User inside Fence

### ISSUE THREAD

**Issue by mhfen** [cowbell/cordova-plugin-geofence#158]
Hi Cowbell Team,

I'm using your plugin in an Ionic 2 application. The plugin works fine in iOS. The plugin is crashing the app on Android any time the addOrUpdate method is called to create a Geofence the user is currently within. 
#### Environment Info:

```
Cordova: 6.2.0
Device: Samsung S6
Android API: 23
Android V: 6
Secondary Device: Nexus 6P
Android API: 23
Android V: 6
```
#### Stacktrace

```
Fatal Exception: java.lang.IllegalStateException: Cannot perform this operation because the connection pool has been closed.
       at android.database.sqlite.SQLiteConnectionPool.throwIfClosedLocked(SQLiteConnectionPool.java:1027)
       at android.database.sqlite.SQLiteConnectionPool.waitForConnection(SQLiteConnectionPool.java:664)
       at android.database.sqlite.SQLiteConnectionPool.acquireConnection(SQLiteConnectionPool.java:397)
       at android.database.sqlite.SQLiteSession.acquireConnection(SQLiteSession.java:905)
       at android.database.sqlite.SQLiteSession.executeForCursorWindow(SQLiteSession.java:834)
       at android.database.sqlite.SQLiteQuery.fillWindow(SQLiteQuery.java:62)
       at android.database.sqlite.SQLiteCursor.fillWindow(SQLiteCursor.java:143)
       at android.database.sqlite.SQLiteCursor.getCount(SQLiteCursor.java:132)
       at android.database.AbstractCursor.moveToPosition(AbstractCursor.java:219)
       at android.database.AbstractCursor.moveToFirst(AbstractCursor.java:258)
       at com.cowbell.cordova.geofence.LocalStorage.getItem(LocalStorage.java:57)
       at com.cowbell.cordova.geofence.GeoNotificationStore.getGeoNotification(GeoNotificationStore.java:20)
       at com.cowbell.cordova.geofence.ReceiveTransitionsIntentService.onHandleIntent(ReceiveTransitionsIntentService.java:69)
       at android.app.IntentService$ServiceHandler.handleMessage(IntentService.java:66)
       at android.os.Handler.dispatchMessage(Handler.java:102)
       at android.os.Looper.loop(Looper.java:158)
       at android.os.HandlerThread.run(HandlerThread.java:61)
```
#### Steps to Reproduce:
1. Launch cordova application on Android device
2. Perform an action which triggers the `geofence.addOrUpdate` method in your code.
3. Observe the application crash
#### Thoughts on a solution

It appears from the stacktrace the crash is kicked off in your LocalStorage.java file on line 53. https://github.com/cowbell/cordova-plugin-geofence/blob/master/src/android/LocalStorage.java#L53

**Comment by mhfen** [comment]
I narrowed this crash to happening only when a fence is created when the user is inside that fence. Updating the bug description.

**Comment by nabinkumarn** [comment]
Has anyone figured out any solution to this? App is crashing when I add multiple geofence and user is already in geofence.

**Comment by bromeostasis** [comment]
Just for reference, it looks like at least @nabinkumarn was able to solve their problem using the suggestion here: https://github.com/cowbell/cordova-plugin-geofence/issues/113#issuecomment-265175493

> I'm looking at removing the database.close() instances as I've found a couple of sites that indicate they aren't necessary!

> http://touchlabblog.tumblr.com/post/24474750219/single-sqlite-connection
http://stackoverflow.com/questions/6608498/best-place-to-close-database-connection

**Comment by tsubik** [comment]
Fixed in #200
